File: src/main/java/easy/ReverseVowels.java
package easy;

public class ReverseVowels {
  public boolean isVowel(char c) {
    return switch (Character.toLowerCase(c)) {
      case 'a', 'e', 'i', 'o', 'u' -> true;
      default -> false;
    };
  }

  public String solve(String s) {
    char[] result = new char[s.length()];
    int left = 0, right = result.length - 1;
    while (left <= right) {

      if (!isVowel(s.charAt(left))) {
        result[left] = s.charAt(left);
        left++;
      } else if (!isVowel(s.charAt(right))) {
        result[right] = s.charAt(right);
        right--;
      } else {
        result[right] = s.charAt(left);
        result[left] = s.charAt(right);
        right--;
        left++;
      }
    }
    return new String(result);
  }
}

File: src/easy/ReverseVowels.swift
func isVowel(_ c: Character) -> Bool {
  switch c.lowercased() {
    case "a", "e", "i", "o", "u":
      return true
    default:
      return false
  }
}

func reverseVowels(_ s: String) -> String {
  var result = Array(s)
  var left = 0
  var right = result.count - 1
  while left <= right {
    if !isVowel(result[left]) {
      left += 1
      continue
    }
    if !isVowel(result[right]) {
      right -= 1
      continue
    }
    result.swapAt(left, right)
    left += 1
    right -= 1
  }

  return String(result)
}
